src/neo4j_graphrag/experimental/components/entity_relation_extractor.py

```diff
@@ -337,3 +344,147 @@ async def run(
+            For each valid node, filter out properties not present in the schema.
+            Remove a node if it ends up with no valid properties.
+            """
+        valid_nodes = []
+        if self.enforce_schema == SchemaEnforcementMode.STRICT:
+            for node in extracted_nodes:
+                if node.label in schema.entities:
+                    schema_entity = schema.entities[node.label]
+                    filtered_props = self._enforce_properties(node.properties,
+                                                              schema_entity)
+                    if filtered_props:
+                        # keep node only if it has at least one valid property
+                        new_node = Neo4jNode(
+                            id=node.id,
+                            label=node.label,
+                            properties=filtered_props,
+                            embedding_properties=node.embedding_properties,
+                        )
+                        valid_nodes.append(new_node)
+        # elif self.enforce_schema == SchemaEnforcementMode.OPEN:
+            # future logic
+        return valid_nodes
+
+    def _enforce_relationships(
+            self,
+            extracted_relationships: List[Neo4jRelationship],
+            filtered_nodes: List[Neo4jNode],
+            schema: SchemaConfig
+    ) -> List[Neo4jRelationship]:
+        """
+        Filter extracted nodes to be conformant to the schema.
+
+        Keep only those whose types are in schema, start/end node conform to schema,
+        and start/end nodes are in filtered nodes (i.e., kept after node enforcement).
+        For each valid relationship, filter out properties not present in the schema.
+        """
+        valid_rels = []
+        if self.enforce_schema == SchemaEnforcementMode.STRICT:
+            valid_node_ids = {node.id for node in filtered_nodes}
+            for rel in extracted_relationships:
+                # keep relationship if it conforms with the schema
+                if rel.type in schema.relations:
+                    if (rel.start_node_id in valid_node_ids and
+                            rel.end_node_id in valid_node_ids):
+                        start_node_label = self._get_node_label(rel.start_node_id,
+                                                                filtered_nodes)
+                        end_node_label = self._get_node_label(rel.end_node_id,
+                                                              filtered_nodes)
+                        if (not schema.potential_schema or
+                                (start_node_label, rel.type, end_node_label) in
+                                schema.potential_schema):
+                            schema_relation = schema.relations[rel.type]
+                            filtered_props = self._enforce_properties(rel.properties,
+                                                                      schema_relation)
```

Comment:
Don't we need to use this here?
```suggestion
                                                                      schema_relation["properties"])
                                                                     
```

Reply:
yea it's a bug, nice catch!